pkg/bff/activity.go

```diff
@@ -0,0 +1,161 @@
+package bff
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/rotationalio/go-ensign"
+	api "github.com/rotationalio/go-ensign/api/v1beta1"
+	"github.com/rotationalio/go-ensign/mock"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/trisacrypto/directory/pkg/bff/models/v1"
+	"github.com/trisacrypto/directory/pkg/store"
+	storeerrors "github.com/trisacrypto/directory/pkg/store/errors"
+	"github.com/trisacrypto/directory/pkg/utils/activity"
+)
+
+// ActivitySubscriber is a struct with a go routine that subscribes to a network
+// activity topic in Ensign and applies asynchronous updates to trtl.
+type ActivitySubscriber struct {
+	client *ensign.Client
+	db     store.Store
+	emock  *mock.Ensign
+	topic  string
+	stop   chan struct{}
+}
```

Comment:
It makes sense to put this in at the BFF top level because of the requirement to access the `store.Store`. I did just want to suggest another pattern for future reference. You could have the subscriber in the `activity` package, and it does all the work of connecting to ensign and subscribing to the topic since the configuration is already in that package -- it can also handle event parsing and sending acks/nacks back. The subscriber takes as input an interface:

```golang
type ActivitySubscriber interface {
     HandleActivity(*NetworkActivity) error 
}
```

`HandleActivity()` is kind of like a callback function (another pattern that could be implemented) - and it is called on each new activity received from the subscriber; if error then a nack is sent back otherwise an ack is sent back.

Reply:
Yeah that might make more sense, I'll keep it in mind.